cleanlab_studio/cli/util.py

```diff
@@ -65,15 +66,34 @@ def is_null_value(val: str) -> bool:
     return val is None or val == "" or pd.isna(val)
 
 
-def init_dataset_from_filepath(filepath: str) -> Dataset:
+def init_dataset_from_filepath(filepath: str) -> Union[Dataset[IO[str]], Dataset[IO[bytes]]]:
     ext = get_dataset_file_extension(filepath)
     if ext == DatasetFileExtension.csv:
         return CsvDataset(filepath)
     elif ext == DatasetFileExtension.xls or ext == DatasetFileExtension.xlsx:
-        return ExcelDataset(filepath)
-    else:  # ext == ".json":
+        return ExcelDataset(filepath, file_type=pathlib.Path(filepath).suffix[1:])
+    elif ext == DatasetFileExtension.json:
         return JsonDataset(filepath)
 
+    raise ValueError(f"filepath {filepath} does not have supported extension.")
+
+
+def init_dataset_from_fileobj(
+    fileobj: Union[IO[str], IO[bytes]], ext: DatasetFileExtension
+) -> Union[Dataset[IO[str]], Dataset[IO[bytes]]]:
+    """Initializes dataset from file object."""
```

Comment:
@axl1313 do these assertions look correct to you?

Reply:
Looks mostly good to me, but might be good to make the type assertions more generic. i.e. `io.TextIOBase` instead of `io.TextIOWrapper` for the csv case so that we can support initializing a dataset from StringIO as well as TextIOWrapper. Up to you whether to make this change though since for now we know we'll only be using this function with TextIOWrapper and BytesIO